images/virtualization-artifact/pkg/controller/vd/internal/source/object_ref.go

```diff
@@ -68,8 +72,19 @@ func (ds ObjectRefDataSource) Sync(ctx context.Context, vd *virtv2.VirtualDisk)
 	condition, _ := service.GetCondition(vdcondition.ReadyType, vd.Status.Conditions)
 	defer func() { service.SetCondition(condition, &vd.Status.Conditions) }()
 
-	if vd.Spec.DataSource.ObjectRef.Kind == virtv2.VirtualDiskObjectRefKindVirtualDiskSnapshot {
+	switch vd.Spec.DataSource.ObjectRef.Kind {
```

Comment:
Ensure that neither DataSource nor ObjectRef are nil.

Reply:
Add.